### DISCUSSION TITLE
Can I display a list of images based on the current user id?

### DISCUSSION THREAD

**Discussion by saidqb** [awcodes/filament-curator#214] [Q&A] [answered]
I want to display a list of images based on the uploaded user.
but I can't find the user id in the table
do I have to add a new field in the migration table?

**Comment by awcodes** [comment]
You would need to set up a relationship between your user and the images.

**Reply by saidqb** [reply]
Am I wrong if I add a column user_id to the media table.

this is what i do now.

2023_06_24_204123_add_user_id_to_media_table.php
```
<?php

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

return new class extends Migration
{
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        Schema::table('media', function (Blueprint $table) {
            $table->unsignedBigInteger('user_id')->default('0')->after('id');
        });
    }

    /**
     * Reverse the migrations.
     *
     * @return void
     */
    public function down()
    {
        Schema::table('media', function (Blueprint $table) {
            $table->dropColumn('user_id');
        });
    }
};

```

in my Model Media

```

       public function newQuery($exclude = true) {
        $data = parent::newQuery($exclude);

        if(auth()->user()){
            if(empty(auth()->user()->getAttribute('role'))) {
                if(auth()->user()->hasRole('Author')){
                    $data->where(['user_id' => auth()->user()->getAttribute('id')]);
                }
            }
        }

        return $data;
    }

    public static function boot()
    {
        parent::boot();

        self::creating(function($model){
            $model->user_id = auth()->user()->id;
        });   
     }
 ```

**Reply by awcodes** [reply, marked as answer]
Not entirely sure I'm following what you are trying to accomplish. Have you looked at the path generators? There's one for storing any images uploaded by a user and scoping those images to a directory for the user's ID. https://github.com/awcodes/filament-curator#path-generation

**Reply by saidqb** [reply]
i will try it better. thanks for a reply

**Comment by pfzwaan** [comment]
I found this thread, I have a question. https://github.com/awcodes/filament-curator#path-generation , will this help me showing images to a certain user_id, so other users won't be able to access their images or delete their images? What will be the best sollution for this?

**Reply by awcodes** [reply]
Yes, use a custom path generator. Then on the picker you can use `->limitToDirectory()` based on the logged in user. You will also need to set up model policies to prevent them from showing up in the media resource.

**Reply by raaquino** [reply]
Can we use the tenant_id in the Media table for this? I tried registering a MediaObserver to store the user_id of the person uploading the media but it does not work.

` if (auth()->check()) {
            $media->tenant_id = auth()->id();
        }`

**Reply by mrdulal** [reply]
Have you solved?
